### ISSUE TITLE
I want disable Reports = No logging (to save memory)

### ISSUE THREAD

**Issue by Thomas499** [blokadaorg/blokada#353]
**Background**
I downloaded blokada on a 2nd generation firestick (not the 4K firestick). Memory on the 2nd generation firestick is a considerable limitation. What i've noticed is that after three or four days the blokada.log is in the higher single digits MB like 8.2MB. 

**Why this is a concern**
Even if you go to app management and clear cache for blokada it doesn't clear the blokada.log which is growing quite fast. At this rate, in less than three months the blokada.log will completely take up all the available memory. And it takes a somewhat advanced user to go into system files to manually delete the file.

**Why I disabled logging**
I personally disabled the reports option on this particular firestick for a few reasons:
1. I know blokada works on firesticks. 
2. The lady had the firestick plugged into the tv for power directly. Meaning when the tv was powered down the firestick has no power. - This means sending the report at a pre-determined time everyday probably isn't going to work in this environment. 
3. If the firestick sent the report soon as the firestick powered up this would be bad. Reason: The second generation firestick already is noticeably slower than the 4k firestick. Opening the apps result in a noticeable lag already. Adding another process that the firestick must go through to send the report, when the person first powers it on with the intention of watching a movie isn't in the persons best interest.
4. The lady was using her phone as a hotspot because she doesn't have an internet service provider for her home. Even if she had unlimited data on her phone they would impose speed caps once threshold is reached. So I don't see a justifiable reason to send reports of the logs everyday when I know blokada works. 

**Speculated possibilities**
It is possible that you have it programmed once the report is sent (which is still unclear if it happens daily automatically or only on the days blokada crashes) that after the report is sent the log is cleared. But if report is disabled then it can't currently clear the blokada.log. But if report is disabled then there is really no reason to log anything. 

**Why only advanced users will benefit from logging** 
In the future some users might want to know what sites were blocked. For example, with goodbye ads by Jerry enabled, mas-ext.amazon.com is blocked which means you can't download any (official) apps on the firestick as documented here  https://github.com/jerryn70/GoodbyeAds/issues/13 If there was an easy way to debug that mas-ext.amazon.com was one of the URL's blocked in the past 3 or so minutes when user was trying to download a new app, that would be very useful to some advanced users who don't have pi hole logging available that they can quickly switch to during the debugging process. But most users are simple creatures who have no business trying to debug anything. These users would benefit from blokada not storing a log so less memory is allocated on their devices. Most of my users just want blokada to work, they don't even care to know blokada is on their devices. 

When their firesticks have a problem (such as memory is out and they can't download a new app) they call me. First I tell them to manually clear cache. Clearing cache will not release the memory allocated by the ever increasing blokada.log file. When clearing cache doesn't work for them I have to personally go over to their home to fix the issue.

**Comment by rpeter85** [comment]
Revoke permission to access external storage.
If you didn't grant, you won't have logs.

**Comment by Thomas499** [comment]
I transferred the blocklist and whitelist from one firestick to the next by copying the filters2.pt file in the io.paperbd folder. Before the changes would take effect I had to go to blacklist->quick actions -> Import/Export->continue->click internal storage first so external storage would be available-> click external storage so import now would be available -> click grant permissions -> click import now -> continue. 

If I didn't enable permissions I believe the transfer of list would not take effect. I don't want to re-type all six custom list urls for the added host list using the firestick remote for each firestick I put blokada on. 

With firesticks if you go to settings->applications->manage installed applications->blokada->permissions it will show this page. 
![image](https://user-images.githubusercontent.com/38442058/55081696-94caa580-505d-11e9-91a8-571f474049d1.png)

It says press select to deny permission, but if you press select it makes a noise and doesn't change. The noise sound like "dling" and insticks tell me it is the sound that you pressed something that isn't allowed. Similar to the sound I would expect if I tried to change an option which required root that I didn't have permissions to change. This is a bug on the firestick side not the blokada side. It might even be specific to Fire OS 6.2.6.0 update that came out two or three days ago . I will report it to the firestick project managers. 

**Relevant questions regarding blokada**

1. If I revoke permission to access external storage will the blacklist/whitelist that I transferred still be used?

2. Is there a way to revoke permission from the app internally? I'm wondering if the firestick permissions page is the only way to revoke permissions or if there is a way I can do it from within the app?

**Comment by rpeter85** [comment]
1.) as soon as the lists are loaded into memory and the configuration saved, Blokada should persist the lists and even if you revoke the permission, the lists should be loaded on restart.
EXCEPT if the lists aren't loaded from an online source, but from the storage on the phone. Then the permission is necessary.

2.) No, apps usually ask for permission (Blokada doesn't do it, only at export/import) and persist that, you need to revoke it on system level

**Comment by Thomas499** [comment]
Thank you for taking the time to answer my questions.

I'm thinking about starting a Github page for my custom list so I can revoke the permission and still keep my custom list. 

Firestick manager responded to the bug report that I filed with them about not being able to change permissions below:

> I have shared you're feedback with the Fire stick product team . Please be assured that you will be receiving a response from them soon as I've pinned the issue to be resolved at the earliest.


I'll let you know when they fix the issue. Closing this issue.
